images/cfw/opt/x1plusd.py

```diff
@@ -1,116 +1,282 @@
+            logger.info("OTA check is disabled, skipping check!")
+            return json.dumps({"ota": ota_response})
+
+        # If we are here we want to check, so check
+        try:
+            # Update check timestamp first
+            self.last_check_timestamp = datetime.datetime.now().timestamp()
+            r = requests.get(self.ota_url, timeout=5)
+            self.last_check_response = r.json()
+        except Exception as e:
+            logger.error(f"Exception calling OTA URL! Error of: {e}")
+            # we Timed out, or hit other error with requests
+            self.last_check_error = True
+            return json.dumps({"ota": ota_response})
+
+        # Now that we have the build info, do our check to see if there's an update
+        if self.build_info.get("buildTimestamp", 0) < self.last_check_response.get(
+            "buildTimestamp", 0
+        ):
+            self.ota_available = True
+
+        # Return result of our check
+        return json.dumps({"ota": ota_response})
+
+
 class SettingsService:
-    DEFAULT_X1PLUS_SETTINGS = { }
+    DEFAULT_X1PLUS_SETTINGS = {
+        "boot.quick_boot": False,
+        "boot.dump_emmc": False,
+        "boot.sdcard_syslog": False,
+        "boot.perf_log": False,
+        "ota.enable": False,
+    }
 
     def __init__(self, router):
         self.router = router
 
         if IS_EMULATING:
+            self.settings_dir = "/"
             self.filename = "/tmp/x1plus-settings.json"
         else:
-            settings_dir = f"/mnt/sdcard/x1plus/printers/{_get_sn()}"
-            self.filename = f"{settings_dir}/settings.json"
-            os.makedirs(settings_dir, exist_ok = True)
-        
+            self.settings_dir = f"/mnt/sdcard/x1plus/printers/{_get_sn()}"
+            self.filename = f"{self.settings_dir}/settings.json"
+            os.makedirs(self.settings_dir, exist_ok=True)
+
         # Before we startup, do we have our settings file? Try to read, create if it doesn't exist.
         try:
-            with open(self.filename, 'r') as fh:
+            with open(self.filename, "r") as fh:
                 self.settings = json.load(fh)
         except FileNotFoundError as exc:
-            logger.warning('settings file does not exist; creating with defaults...')
-            # TODO: Add logic here (call helper function) to check for flag files, 
-            # and adjust our defaults to match
-            self.settings = copy.deepcopy(SettingsService.DEFAULT_X1PLUS_SETTINGS)
+            logger.warning("settings file does not exist; creating with defaults...")
+            self.settings = self._migrate_old_settings()
             self._save()
-    
+
+    def _migrate_old_settings(self):
+        """
+        Used to migrate init.d flag files to our new json on first run.
+        All bbl_screen settings MUST BE MIGRATED BY bbl_screen!
+        """
+        defaults = copy.deepcopy(SettingsService.DEFAULT_X1PLUS_SETTINGS)
```

Comment:
maybe it should just be `{}` as the default no matter what?  the default is False after all for absent things

Reply:
idea is to also have a source of truth for defaults in one spot at file generation, vs defaults all over the place in bbl_screen.